components/SelectorBase/SelectorBase.vue

```diff
@@ -247,6 +291,81 @@ export default {
     handleDelete() {
       this.selectedOption = '';
     },
+    removeElements(page) {
+      try {
+        page.getElementById('toolbar-administration').remove();
+        page
+          .getElementsByClassName('gin-secondary-toolbar layout-container')[0]
+          .remove();
+        page.getElementsByTagName('header')[0].style.top = 0;
+        page.getElementsByClassName('gin--vertical-toolbar')[0].className = '';
+        page
+          .getElementById('block-farm-powered')
+          .getElementsByTagName('a')[0]
+          .removeAttribute('href');
+      } catch (e) {
+        // couldn't remove elements
+      }
+    },
+    checkValidUrl(iframe) {
+      try {
+        const iframeUrl = iframe.contentWindow.location.href;
+        // prohibits navigating away from the add form
+        if (!iframeUrl.includes(this.popupUrl)) {
+          this.hidePopup();
+        }
+      } catch (e) {
+        this.hidePopup();
+      }
+    },
+    getNewAsset(document) {
+      let result = null;
+      try {
+        result = document
+          .getElementsByClassName('messages-list')[1]
+          .getElementsByClassName('messages__wrapper')[0]
+          .getElementsByClassName(
+            'messages-list__item messages messages--status'
+          )[0]
+          .getElementsByClassName('messages__content')[0]
+          .getElementsByClassName('placeholder')[0].innerText;
+        return result;
+      } catch (e) {
+        // couldn't get result
+        return result;
+      }
+    },
+    showPopup() {
+      this.popupSrc = this.popupUrl;
+      this.isPopupVisible = true;
+    },
+    hidePopup() {
+      this.isPopupVisible = false;
+      this.popupSrc = '';
+      this.$emit('add-clicked', '');
+    },
+    handleIFrameLoad() {
+      const iframe = document.getElementById('popupIframe');
+
+      // check that the current Url is valid
+      this.checkValidUrl(iframe);
+
+      // show page after removing elements
+      this.removeElements(iframe.contentWindow.document);
+      this.isLoaded = true;
+
+      // when the iframe redirects, hide it
+      iframe.contentWindow.onunload = () => {
+        this.isLoaded = false;
+      };
+
+      // attempts to get & fill-in the newly created asset
+      let result = this.getNewAsset(iframe.contentWindow.document);
+      if (result) {
+        this.hidePopup();
+        this.$emit('add-clicked', result.trim());
```

Comment:
The emit should have a JSdoc comment describing the emitted event and its payload.  See some of the other component for examples.  The docs should be clear about what the payload is if the popup is closed without having created the asset.